[For Net9.0 TFM] Change MicrosoftExtensionsLoggingAbstractions from 8.0.2 to 8.0.0

mdchennu opened the pull request:
# [For Net9.0 TFM] Change Abstractions from 8.0.2 to 8.0.0
<!-- Thank you for submitting a pull request to our repo. -->

<!-- If this is your first PR in the IdentityModel repo, please run through the checklist
below to ensure a smooth review and merge process for your PR. -->

- [X] You've read the [Contributor Guide](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/Contributing.md) and [Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
- [X] You've included unit or integration tests for your change, where applicable.
- [X] You've included inline docs for your change, where applicable.
- [X] If any gains or losses in performance are possible, you've included benchmarks for your changes. [More info](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki/Benchmarking-in-Wilson)
- [X] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

<!-- Once all that is done, you're ready to go. Open the PR with the content below. -->

## Description



This PR updates Msft.Extensions.Logging.Abstractions to 8.0.0 for .Net9.0

**Covered in PR**
Update Microsoft.Extensions.Logging.Abstractions to 8.0.0 for .NET9.0 from 8.0.2
Fixes #{[3225](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/3225)} (in this specific format)

For testing:
1. Built the solution -> clean build for all .NET9 Frameworks
2. Ran the tests -> all passed

Commit:
removed cs proj reference to System.Text.Json and updated Abstractions

kllysng commented on src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj:
```diff
@@ -57,7 +57,7 @@
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp' And !$(TargetFramework.StartsWith('net9.0'))">
```
Our NetCoreApp target fw's are [net6, net8, and net9](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/35caa6cb0b693188b5c5c1b9434414f1a3ba43ba/build/targets.props#L3C83-L3C104). Consider being explicit like:  
```suggestion
  <ItemGroup Condition="$(TargetFramework) == 'net6.0' or $(TargetFramework) == 'net8.0'">
```

mdchennu replied:
thank you! I've updated!

mdchennu replied:
Wait, Kelly, I tried changing the condition to   `<ItemGroup Condition="'$(TargetFramework)'== 'net6.0' or '$(TargetFramework)' == 'net8.0'">: `

Then I got a bunch of build errors for the three `csproj` files updated with the condition above. The build errors were from all TFMs and were of two categories:

1. cannot find dlls
2. `The type or namespace name 'JsonIgnore' could not be found (are you missing a using directive or an assembly reference?`

I tried researching why and found this thru copilot hah: 
`The error occurs because in .NET 9.0, even though System.Text.Json is included as part of the framework, you need to add an explicit reference to the assembly through a using directive. This is due to how JsonIgnore and other Json-related attributes are organized in .NET.`

The explicit reference I added to the csproj files was:
```
<ItemGroup Condition="'$(TargetFramework)' == 'net9.0'">
  <FrameworkReference Include="Microsoft.AspNetCore.App" />
</ItemGroup>
```

This change still caused build errors similar to above, so now I updated the reference as follows: 
```
  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp' And '$(TargetFramework)' != 'net9.0' ">
    <PackageReference Include="System.Text.Json" Version="$(SystemTextJson)" />
  </ItemGroup>
```

What do you think?

kllysng replied:
Hi Mallika! I pulled your branch and tried things out. Sorry, I misread the initial line and interpreted it as `$(TargetFramework) == NetCoreApp` instead of `!=`. 

You can remove the second half of the statement because any fw that's != NetCoreApp won't start with net9.0 and I've verified it locally

mdchennu replied:
All good! thank you! I removed it

mdchennu committed:
Merge branch 'dev' into mallikac/remove-system-text-json-and-update-abstractions

Commit:
updated with PR comments

Commit:
Merge branch 'mallikac/remove-system-text-json-and-update-abstractions' of https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet into mallikac/remove-system-text-json-and-update-abstractions

kllysng commented on src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj:
```diff
@@ -57,7 +57,7 @@
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
+  <ItemGroup Condition="$(TargetFramework) == 'net6.0' or $(TargetFramework) == 'net8.0'">
```
Sorry I missed this in my last comment, here and elsewhere, we also need single quotes around $(TargetFramework)
```suggestion
  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0' or '$(TargetFramework)' == 'net8.0'">
```

mdchennu committed:
Merge branch 'dev' into mallikac/remove-system-text-json-and-update-abstractions

Commit:
updated statement to be more clear

Commit:
Merge branch 'mallikac/remove-system-text-json-and-update-abstractions' of https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet into mallikac/remove-system-text-json-and-update-abstractions

brentschmaltz commented on src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Microsoft.IdentityModel.Protocols.OpenIdConnect.csproj:
```diff
@@ -28,10 +28,14 @@
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'net9.0'">
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
```
asp.net takes a dependency on M.IM.Protocols.OpenIdConnect, is there a circular reference here?

pmaytak replied:
Why is this needed?

brentschmaltz commented on src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj:
```diff
@@ -57,7 +57,7 @@
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp' And '$(TargetFramework)' != 'net9.0' ">
```
How will M.IM.Tokens use System.Text.Json?
For example, JsonSerializerPrimitives references Utf8JsonReader.

brentschmaltz reviewed (changes requested):
couple of comments about circular reference and System.Text.Json used by M.IM.Tokens.

pmaytak commented on build/dependencies.props:
```diff
@@ -19,9 +19,10 @@
       Do not bump to 9.0. complicated reasons due to some runtime incompatibilities if 
-      CoreWCF compiles against an older version but then run with a newer version in a related package
+      CoreWCF compiles against an older version but then run with a newer version in a related package.
+      Also, moved to 8.0.0 due to removal of Microsoft.Bcl.Memory 9.0 in Wilson
```
What does this mean? Why is Bcl.Memory removal relevant to this change?

mdchennu replied:
Okay, I spoke to Sruthi! Bcl.Memory is not relevant to this change. It was included in the bug description because the customer left this feedback in the same github issue:

![image](https://github.com/user-attachments/assets/8724de6b-29c4-4a76-9383-b03a611cc344)

Link: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/3208

I think this change should still go in, but I'll remove the comments that relate it to Bcl.Memory work. What do you think? @sruke @pmaytak

sruke replied:
Makes sense to me.

pmaytak commented on src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Microsoft.IdentityModel.Protocols.OpenIdConnect.csproj:
```diff
@@ -28,10 +28,14 @@
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp' And '$(TargetFramework)' != 'net9.0' ">
```
.NETCoreApp already includes net9.0.

pmaytak commented:
I don't think we need to do anything extra related to System.Text.Json. It's already only used for non-.NET Core TFMs.

https://www.nuget.org/packages/Microsoft.IdentityModel.Protocols.OpenIdConnect/8.9.0#dependencies-body-tab

mdchennu commented:
> I don't think we need to do anything extra related to System.Text.Json. It's already only used for non-.NET Core TFMs.
> 
> https://www.nuget.org/packages/Microsoft.IdentityModel.Protocols.OpenIdConnect/8.9.0#dependencies-body-tab

Okay, so I was looking at the comments in the Bug description and it mentioned to make the following two changes (which I did in the PR):
![image](https://github.com/user-attachments/assets/d7cd2686-94e4-4422-b0d7-9aef231c09d5)

Bug description: https://identitydivision.visualstudio.com/Engineering/_workitems/edit/3238459

Are both changes (highlighted in green) still necessary? @sruke @pmaytak

Commit:
remove the AspNetCore reference

mdchennu commented:
> couple of comments about circular reference and System.Text.Json used by M.IM.Tokens.

Thank you for the comments! I've removed the circular reference, and removed the change I added in for usage of System.Text.Json. I'm going to ask Sruthi and Peter about if this change is still required (I think I'm having a hard time understanding why we need to make these changes for Bcl.Memory removal), and then follow up on this PR.

pmaytak commented:
> > I don't think we need to do anything extra related to System.Text.Json. It's already only used for non-.NET Core TFMs.
> > https://www.nuget.org/packages/Microsoft.IdentityModel.Protocols.OpenIdConnect/8.9.0#dependencies-body-tab
> 
> Okay, so I was looking at the comments in the Bug description and it mentioned to make the following two changes (which I did in the PR): ![image](https://private-user-images.githubusercontent.com/45950356/443286426-d7cd2686-94e4-4422-b0d7-9aef231c09d5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcxNTQ2ODgsIm5iZiI6MTc0NzE1NDM4OCwicGF0aCI6Ii80NTk1MDM1Ni80NDMyODY0MjYtZDdjZDI2ODYtOTRlNC00NDIyLWIwZDctOWFlZjIzMWMwOWQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTEzVDE2Mzk0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhYWFhZjBlYjdkODVmZDg3YjUxYzJlNTQ2ZDkxZjdmZTliZmYyZmM2NzA0MzgyMTg5MDM4MTNlY2NlZjA5YzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fNCLoZH2rQIh7cJPcoEAWAQMp5P4m-WmOM9VsudQs-s)
> 
> Bug description: https://identitydivision.visualstudio.com/Engineering/_workitems/edit/3238459
> 
> Are both changes (highlighted in green) still necessary? @sruke @pmaytak

Related to System.Text.Json, the work item may just be incorrect. I took a look at a few Wilson packages on Nuget and none reference STJ for newer frameworks (correctly).

Commit:
merge from dev

Commit:
updated comment
